### PR TITLE
Add labels on reminders

### PR DESCRIPTION
**Pull request by PaoloRotolo**
Please review and merge.

### CODE REVIEW

File: app/src/main/java/org/glucosio/android/activity/RemindersActivity.java [hunk 1]
```diff
@@ -26,11 +31,13 @@
     private FloatingActionButton addFab;
     private RemindersPresenter presenter;
     private ListView listView;
+    private String label;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_reminders);
+        final Activity ReminderActivity = this;
```
Comment (emartynov) [inline app/src/main/java/org/glucosio/android/activity/RemindersActivity.java]: While Capitalised?

File: app/src/main/java/org/glucosio/android/activity/RemindersActivity.java [hunk 2]
```diff
@@ -46,18 +53,38 @@ protected void onCreate(Bundle savedInstanceState) {
...
-                TimePickerDialog tpd = TimePickerDialog.newInstance(
-                        RemindersActivity.this,
-                        cal.get(Calendar.HOUR_OF_DAY),
-                        cal.get(Calendar.MINUTE),
-                        is24HourFormat);
-                tpd.show(getFragmentManager(), "Timepickerdialog");
-            }
+            public void onClick(final View view) {
+
+                AlertDialog.Builder builder = new AlertDialog.Builder(ReminderActivity);
```
Comment (emartynov) [inline app/src/main/java/org/glucosio/android/activity/RemindersActivity.java]: Should we separate text and time clicking? It is super inconvenient to edit it in two steps

File: app/src/main/java/org/glucosio/android/db/Migration.java [hunk 3]
```diff
@@ -169,7 +169,8 @@ public void apply(DynamicRealmObject obj) {
                     .addField("metric", String.class)
                     .addField("alarmTime", Date.class)
                     .addField("active", Boolean.class, FieldAttribute.REQUIRED)
-                    .addField("oneTime", Boolean.class, FieldAttribute.REQUIRED);
+                    .addField("oneTime", Boolean.class, FieldAttribute.REQUIRED)
+                    .addField("label", String.class);
```
Comment (emartynov) [inline app/src/main/java/org/glucosio/android/db/Migration.java]: We don't do db version upgrade because it is not release yet. Correct?

Comment (PaoloRotolo) [inline app/src/main/java/org/glucosio/android/db/Migration.java]: Yes, it's still in the daily channel

File: app/src/main/java/org/glucosio/android/activity/RemindersActivity.java (hunk 2 above)
Comment (PaoloRotolo) [inline app/src/main/java/org/glucosio/android/activity/RemindersActivity.java]: IMHO it looks bad to have just 2 EditText in one dialog. And you still need to open the time picker dialog anyway.

File: app/src/main/java/org/glucosio/android/activity/RemindersActivity.java (hunk 1 above)
Comment (PaoloRotolo) [inline app/src/main/java/org/glucosio/android/activity/RemindersActivity.java]: Typo

**Comment by PaoloRotolo** [comment]
About notification I'm waiting for @bkerensa to give me the final Array of strings.

File: app/src/main/java/org/glucosio/android/activity/RemindersActivity.java (hunk 2 above)
Comment (emartynov) [inline app/src/main/java/org/glucosio/android/activity/RemindersActivity.java]: How about two touch spots on Reminder item in the list?

File: app/src/main/java/org/glucosio/android/tools/GlucosioNotificationManager.java [hunk 4]
```diff
@@ -20,8 +20,8 @@ public GlucosioNotificationManager(Context context) {
         this.context = context;
     }
 
-    public void sendReminderNotification() {
-        String NOTIFICATION_TITLE = "\u23f0";
+    public void sendReminderNotification(String label) {
+        String NOTIFICATION_TITLE = label + " " + "\u23f0";
```
Comment (emartynov) [inline app/src/main/java/org/glucosio/android/tools/GlucosioNotificationManager.java]: Should not be capitalised

**Comment by bkerensa** [comment]
@PaoloRotolo @emartynov what is the status with this? We need to land this sooner than later so we can begin testing up till our next release in October.

**Comment by bkerensa** [comment]
@PaoloRotolo we need the strings added soon and to land this

**Comment by PaoloRotolo** [comment]
Okay it's ready to merge right now.

File: app/src/main/java/org/glucosio/android/tools/GlucosioNotificationManager.java [hunk 5]
```diff
@@ -68,4 +71,9 @@ public void sendReminderNotification() {
         NotificationManagerCompat notificationManagerCompat = NotificationManagerCompat.from(context);
         notificationManagerCompat.notify(NOTIFICATION_ID, notification);
     }
+
+    private int generateRandomNumber(int min, int max) {
+        Random r = new Random();
```
Comment (emartynov) [inline app/src/main/java/org/glucosio/android/tools/GlucosioNotificationManager.java]: I would say `Random` is extra here. You can just use `System.currentMillesecs() % 2`

File: app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -449,10 +449,19 @@
...
     <string name="activity_reminders_title">Reminders</string>
-    <string name="reminders_notification_text">Time to log your glucose level!</string>
     <string name="reminders_notification_action">Add reading</string>
     <string name="glucosio_activity_reminders_empty">No reminders set. Add a new reminder here.</string>
     <string name="activitiy_reminders_error_duplicate">This reminder already exists.</string>
     <string name="activity_reminder_error_empty">Please enter a valid label.</string>
     <string name="activity_reminder_add_label">Set reminder Label</string>
+    <string name="reminder_item_title_morning">Good Morning, It is time to do an entry in Glucosio!</string>
+    <string name="reminder_item_title_afternoon">Good Afternoon, It is time to do an entry in Glucosio!</string>
+    <string name="reminder_item_title_evening">Good Morning, It is time to do an entry in Glucosio!</string>
```
Comment (bkerensa) [inline app/src/main/res/values/strings.xml]: @PaoloRotolo

File: app/src/main/java/org/glucosio/android/tools/GlucosioNotificationManager.java [hunk 7]
```diff
@@ -46,14 +49,14 @@ public void sendReminderNotification() {
             notification = new Notification.Builder(context)
                     .setContentTitle("\u23f0")
                     .setContentText(NOTIFICATION_TEXT)
-                    .setSmallIcon(R.drawable.ic_stat_glucosio)
+                    .setSmallIcon(R.drawable.ic_stat_glucfosio)
```
Comment (bkerensa) [inline app/src/main/java/org/glucosio/android/tools/GlucosioNotificationManager.java]: @PaoloRotolo

**Comment by bkerensa** [comment]
We have just a couple weeks left of opportunity to test this before next release! Ideally we want this out before OSCON

**Comment by emartynov** [comment]
It is ready to be merged one fix about the "Morning..." text that should "Evening..."

File: app/src/main/java/org/glucosio/android/tools/GlucosioNotificationManager.java (hunk 7 above)
Comment (emartynov) [inline app/src/main/java/org/glucosio/android/tools/GlucosioNotificationManager.java]: Not a problem

File: app/src/main/res/values/strings.xml (hunk 6 above)
Comment (emartynov) [inline app/src/main/res/values/strings.xml]: Fixed

**Comment by emartynov** [comment]
I think, it is ready to merge

**Pull request merged**